etna/core/models/mixins.py

```diff
@@ -25,3 +30,47 @@ class ContentWarningMixin(models.Model):
 
     class Meta:
         abstract = True
+
+
+class NewLabelMixin(models.Model):
+    """Mixin to allow editors to toggle 'new' label to be applied on-publish"""
+
+    mark_new_on_next_publish = models.BooleanField(
+        verbose_name="mark this page as 'new' when published",
+        default=True,
+    )
+
+    newly_published_at = models.DateField(
+        editable=False,
+        default=None,
+        null=True,
+    )
+
+    new_label_display_for_days = 21
+
+    def save(self, *args, **kwargs):
+        if self.live and self.mark_new_on_next_publish:
+            self.mark_new_on_next_publish = False
+            self.newly_published_at = timezone.now().date()
+        return super().save(*args, **kwargs)
```

Comment:
@jamesbiggs I think all we're missing now is the bit that sets `mark_new_on_next_publish` to `False` in the `latest_revision.content` after saving. Without this, that checkbox will remain checked the next time the page is edited (Wagtail always loads the latest revision content into the page editor). We want that checkbox to be unchecked after the 'marking as new' has happened - then, for editors to have to deliberately find that box and check it again if they need to reset the timer.

Reply:
Hi Andy, sorry it took me a while to get back to you - I had to sort out some issues when pulling the latest develop in with the migrations etc.

Is this not what we wanted? Currently, the editor can save the page as draft with the box ticked - this will then remain ticked, and the date won't be set.

When the page is published, the date is set, and the box goes back to the un-ticked state and stays that way.

From what I'm understanding, the `latest_revision.content` suggestion would make it so that any time an editor _opens_ the edit page, it'll set the box to be un-ticked?

Reply:
@jamesbiggs The journey you describe is correct, and we're catering for that. However, you're just missing one crucial point, and that is: When you click to edit a page in Wagtail, it's not the saved state of the page you see loaded into the editor.... it's the `latest_revision` content. So, if `mark_new_on_next_publish` value remains True there, the checkbox will also be checked when you next come to edit the page.